packages/frontend/component/src/components/block-suite-editor/index.tsx

```diff
@@ -129,6 +144,42 @@ const BlockSuiteEditorImpl = (props: EditorProps): ReactElement => {
     };
   }, [editor, page.awarenessStore, page.meta.trash, setBlockHub]);
 
+  //hack: scroll to block and select it when jumping to block
+  // {
+  const [selectPath, setSelectPath] = useState<string[]>([]);
+  const [blockElement, setBlockElement] = useState<BlockElement | null>(null);
+  const blockId = useRouterHash();
+
+  if (!blockElement && blockId) {
+    const element = document.querySelector(
+      `[data-block-id="${blockId}"]`
+    ) as BlockElement | null;
+
+    if (element) {
+      setBlockElement(element);
+    }
+  }
```

Comment:
This is effectively a while loop in every render, until blockElement is being set.

Reply:
Could you please tell me how to optimise?

Reply:
I am rewriting this component. The basic idea is to iterate for a few ms until the block element is found ... I guess this is not ideal either, but should be better than relying on render to set the blockElement